### PR TITLE
Release Automation

### PR DESCRIPTION
**Pull request by accraze**
This PR adds the ability to deploy releases to PyPI via TravisCI.

This is done using a script (`deploy.sh`) which will build the binary and wheels and upload via [twine](https://pypi.org/project/twine/) whenever the version gets incremented in `about.py`.

PYPI Credentials will be stored as private environment variables on the TravisCI repo.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,3 +37,13 @@ notifications:
     on_failure: change
     template:
       - "%{repository_slug}#%{build_number} (%{branch} - %{commit} : %{author}): %{message} %{build_url}"
+
+deploy:
+  provider: pypi
+  user: "foobar"
```
Comment (accraze) [inline .travis.yml]: **TODO**: we'll need to encrypt pypi auth and update these vars. 
see: https://docs.travis-ci.com/user/deployment/pypi/

Comment (halfak) [reply .travis.yml]: What would be a good way to do this safely?

Comment (halfak) [reply .travis.yml]: Aha!  Just started reading what you linked to ^_^

**Comment by halfak** [comment]
I'm not sure I understand how this is going to check for a change to the version number.

**Comment by halfak** [comment]
I guess maybe the --skip-existing enforces that?

**Comment by accraze** [comment]
@halfak yep thats correct, the `--skip-existing` flag will enforce only uploading new versions to PyPI.

**Pull request merged**